sarsCoV2Analysis.wdl

```diff
@@ -0,0 +1,472 @@
+  }
+
+  call qcStats {
+    input:
+      sample = samplePrefix,
+      bam = select_first([articTrimming.sortedBam, bowtie2Sensitive.bamFile]),
+      hostMappedBam = bowtie2HumanDepletion.hostMappedBam
+  }
+
+  call blast2ReferenceSequence {
+    input:
+      consensusFasta = variantCalling.consensusFasta
+  }
+
+  call spadesGenomicAssembly {
+    input:
+      fastq1 = bowtie2HumanDepletion.out1,
+      fastq2 = bowtie2HumanDepletion.out2,
+      sample = samplePrefix
+  }
+
+  output {
+    File hostRemovedR1Fastq = bowtie2HumanDepletion.out1
+    File hostRemovedR2Fastq = bowtie2HumanDepletion.out2
+    File hostMappedBam = bowtie2HumanDepletion.hostMappedBam
+    File hostMappedBai = bowtie2HumanDepletion.hostMappedBai
+    File taxonomicClassification = kraken2.out
+    File bam = bowtie2Sensitive.bamFile
+    File bai = bowtie2Sensitive.baiFile
+    File? primertrimSortedBam = articTrimming.sortedBam
+    File? primertrimSortedBai = articTrimming.sortedBai
+    File? cvgHist = articTrimming.cvgHist
+    File vcf = variantCalling.vcfFile
+    File consensusFasta = variantCalling.consensusFasta
+    File variantOnlyVcf = variantCalling.variantOnlyVcf
+    File bl2seqReport = blast2ReferenceSequence.bl2seqReport
+    File genomecvgHist = qcStats.genomecvgHist
+    File genomecvgPerBase = qcStats.genomecvgPerBase
+    File hostMappedAlignmentStats = qcStats.hostMappedAlignmentStats
+    File hostDepletedAlignmentStats = qcStats.hostDepletedAlignmentStats
+    File spades = spadesGenomicAssembly.sampleSPAdes
+  }
+}
+
+task bbMap {
+  input {
+    String modules = "bbmap/38.75"
+    String bbMap = "bbmap"
+    File fastq1
+    File fastq2
+    String sample
+    String reference = "$BBMAP_ROOT/share/bbmap/resources/adapters.fa"
+    Int trimq = 25
+    Int mem = 8
+    Int timeout = 72
+  }
+
+  command <<<
+    set -euo pipefail
+
+    bbmap bbduk in1=~{fastq1} in2=~{fastq2} \
+    out1=~{sample}_qad_r1.fastq.gz out2=~{sample}_qad_r2.fastq.gz \
+    ref=~{reference} \
+    ktrim=r k=23 mink=11 hdist=1 tpe tbo qtrim=rl trimq=~{trimq}
+  >>>
+
+  runtime {
+    memory: "~{mem} GB"
+    modules: "~{modules}"
+    timeout: "~{timeout}"
+  }
+
+  output {
+    File out1 = "~{sample}_qad_r1.fastq.gz"
+    File out2 = "~{sample}_qad_r2.fastq.gz"
+  }
+}
+
+task bowtie2HumanDepletion {
+  input {
+    String modules = "bowtie2/2.3.5.1 samtools/1.9 hg38-bowtie-index/2.3.5.1"
+    File fastq1
+    File fastq2
+    String reference = "$HG38_BOWTIE_INDEX_ROOT/hg38_random_index"
+    String sample
+    Int mem = 12
+    Int timeout = 72
+  }
+
+  String hostMappedSam = "~{sample}.host.mapped.sam"
+  String hostMappedBam_ = "~{sample}.host.mapped.bam"
+  String hostMappedBai_ = "~{sample}.host.mapped.bam.bai"
+
+  command <<<
+    set -euo pipefail
+
+    bowtie2 --quiet -x ~{reference} \
+    -1 ~{fastq1} -2 ~{fastq2} \
+    --un-conc-gz ~{sample}_host_removed_r%.fastq.gz \
+    -S ~{hostMappedSam}
```

Comment:
Could pipe `bowtie2` output directly into `samtools view -b` here -- faster and eliminates writing a SAM file to temporary space